### PR TITLE
fix: kv default traversal

### PR DESCRIPTION
**Pull request by cristianmtr**

### CODE REVIEW

**Review by JoanFM — COMMENTED** [review]
@maximilianwerk I think this is the value we should put as default, do you agree?

**Comment by maximilianwerk** [comment]
> @maximilianwerk I think this is the value we should put as default, do you agree?

Yes.

**Review by maximilianwerk — DISMISSED** [review]
LGTM

**Comment by cristianmtr** [comment]
Some test was failing

**Review by florian-hoenicke — DISMISSED** [review]
Nice feature. For most of the real world cases it will make the yaml configuration smaller.
The test fails because there is a `traversal_paths: ['r']` missing on the `SearchRequest` in `tests/integration/crud/yaml/index_kv.yml`.
The test `tests/integration/crud/test_crud.pytest_delete_kv:test_crud.py` fails because it is just using the `kv indexer` and tries to get matches.

**Comment by cristianmtr** [comment]
> Nice feature. For most of the real world cases it will make the yaml configuration smaller.
> The test fails because there is a `traversal_paths: ['r']` missing on the `SearchRequest` in `tests/integration/crud/yaml/index_kv.yml`.
> The test `tests/integration/crud/test_crud.pytest_delete_kv:test_crud.py` fails because it is just using the `kv indexer` and tries to get matches.

Strange, I cannot reproduce the second one. It passes for me. I fixed the first one and started a new run of the CI. Let's see...

**Comment by cristianmtr** [comment]
The test `tests/integration/test_helloworld.py::test_helloworld_py` keeps failing in CI, but cannot reproduce locally

File: tests/unit/drivers/test_kv_search_driver.py [hunk 1]
```diff
@@ -42,10 +42,11 @@ def __init__(self, *args, **kwargs):
             self.db[int(doc.id)] = doc.SerializeToString()
 
 
-
 class SimpleKVSearchDriver(KVSearchDriver):
 
     def __init__(self, *args, **kwargs):
+        if kwargs.get('traversal_paths') is None:
```
Comment (JoanFM) [inline tests/unit/drivers/test_kv_search_driver.py:48]: add `traversal_paths` to set of arguments

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: tests/unit/drivers/test_kv_search_driver.py [hunk 2]
```diff
@@ -42,11 +42,10 @@ def __init__(self, *args, **kwargs):
             self.db[int(doc.id)] = doc.SerializeToString()
 
 
-
 class SimpleKVSearchDriver(KVSearchDriver):
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    def __init__(self, traversal_paths=['r'], *args, **kwargs):
```
Comment (cristianmtr) [inline tests/unit/drivers/test_kv_search_driver.py:47]: @JoanFM like so?

Comment (JoanFM) [reply tests/unit/drivers/test_kv_search_driver.py:47]: yes

**Review by JoanFM — DISMISSED** [review]

**Review by theUnkownName — APPROVED** [review]

File: tests/integration/test_helloworld.py [hunk 3]
```diff
@@ -91,7 +91,7 @@ def test_helloworld_flow(tmpdir):
 
     # run it!
     with f:
-        f.index(_input_ndarray(targets['index']['data']), batch_size=args.index_batch_size)
+        f.index(_input_ndarray(targets['index']['data']), batch_size=args.index_request_size)
```
Comment (JoanFM) [inline tests/integration/test_helloworld.py:94]: can u call it `request_size=`?

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: tests/integration/test_helloworld.py (hunk 3 above)
Comment (cristianmtr) [reply tests/integration/test_helloworld.py]: done

**Review by JoanFM — APPROVED** [review]

**Pull request merged**